File: P/js/code.js
class Vertex {
    constructor(left, right) {
        this.left = left;
        this.right = right;
    }
}

function getTreeBorder(tree, root) {
    // your code goes here
    return [];
}

const _readline = require('readline');

const _reader = _readline.createInterface({
    input: process.stdin
});

const _inputLines = [];
let _curLine = 0;

_reader.on('line', line => {
    _inputLines.push(line);
});

process.stdin.on('end', solve);

function readTree(n) {
    const tree = [];
    for (let i = 0; i < n; i++) {
        const leftAndRight = readArray();
        tree.push(new Vertex(leftAndRight[0], leftAndRight[1]));
    }
    return tree;
}

function solve() {
    const firstLine = readArray();
    const n = firstLine[0];
    const root = firstLine[1];
    const tree = readTree(n);
    outputAnswer(getTreeBorder(tree, root));
}

function readInt() {
    const n = Number(_inputLines[_curLine]);
    _curLine++;
    return n;
}

function readArray() {
    var arr = _inputLines[_curLine].trim(" ").split(" ").map(num => Number(num));
    _curLine++;
    return arr;
}

function outputAnswer(border) {
    process.stdout.write(`${border.join(' ')}`);
}

File: P/cpp/code.cpp
#include <iostream>
#include <vector>

using namespace std;

struct Vertex {
    int left;
    int right;

    Vertex(int left, int right) : left(left), right(right) {
    }
};


vector<int> getTreeBorder(const vector<Vertex>& tree, int root) {
    // your code goes here
    return {};
}

void outputAnswer(const vector<int>& treeBorder) {
    for (int elem : treeBorder) {
        cout << elem << " ";
    }
    cout << endl;
}

vector<Vertex> readTree(int n) {
    vector<Vertex> tree;
    for (int i = 0; i < n; i++) {
        int left, right;
        cin >> left >> right;
        tree.push_back(Vertex(left, right));
    }
    return tree;
}

int main() {
        int n;
        cin >> n;
        int root;
        cin >> root;
        vector<Vertex> tree = readTree( n);
        outputAnswer(getTreeBorder(tree, root));
}

